### PR TITLE
Get Claim By ID Endpoint

### PR DESCRIPTION
**Pull request by kjduensing**
## Summary
In order to allow the claim detail page to show a single specific claim, we need to add a vets-api endpoint to get a claim by claim ID.

## Related issue(s)
Closes department-of-veterans-affairs/va.gov-team#91011

## Testing done
- [x] *New code is covered by unit tests*

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

### CODE REVIEW

File: modules/travel_pay/app/services/travel_pay/claims_service.rb [hunk 1]
```diff
@@ -14,6 +14,21 @@ def get_claims(current_user)
       }
     end
 
+    def get_claim_by_id(current_user, claim_id)
```
Comment (liztownd) [inline modules/travel_pay/app/services/travel_pay/claims_service.rb:17]: 🤔 This is making me rethink the whole refactor to include the tokens also as params.... At what point is it too many params to pass to a function call?

Comment (kjduensing) [reply modules/travel_pay/app/services/travel_pay/claims_service.rb:17]: I should have said earlier, but I don't plan on merging these until you've merged yours, so the burden is on my PRs to fix merge conflicts and figure out that stuff.

That said, I don't like any more than 3 arguments to a method. There are options:
1. dump all the necessary params into an `options` argument. This is sorta low-brow and it hides the level at which the arguments are needed
2. Create the clients at the controller level, constructor inject them into the services. Now there's no need to pass the tokens, and we can focus on passing query/path params. Probably the OO way to do it, but not completely thought out yet

Comment (kjduensing) [reply modules/travel_pay/app/services/travel_pay/claims_service.rb:17]: we can maybe skip the DI stuff and instantiate the clients in their respective services as instance variables.

**Review by liztownd — APPROVED** [review]

**Review by ryan-mcneil — DISMISSED** [review]

**Review by liztownd — APPROVED** [review]
Looks good!

**Review by ryan-mcneil — APPROVED** [review]
👍

**Pull request merged**